ocaml/gencert/lib.ml

```diff
@@ -57,6 +57,58 @@ let validate_private_key pkcs8_private_key =
      )
   >>= ensure_key_length
 
+(* TODO gencert shouldn't have anything to do with CA certificates.
+ * This file should probably become a library with generally useful functions
+ * for operating on certificates *)
+type cert_kind = CA | Server
+
+let pem_of_string ~kind x =
```

Comment:
Instead of using variant, it might be nicer to add the error parameter as
```ocaml
let pem_of_string ~kind x invalid_error =
```
Same with assert_not_expired:
```ocaml
let assert_not_expired ~kind ~now error_not_yet error_expired certificate =
```

Reply:
I agree this is probably the right thing to do, at the cost of an awkward interface